lib/rubocop/cop/style/string_hash_keys.rb

```diff
@@ -30,6 +41,14 @@ def autocorrect(node)
             corrector.replace(node.source_range, symbol_content)
           end
         end
+
+        private
+
+        def environments?(node)
+          node.each_ancestor.any? do |ancestor|
```

Comment:
You might be able to avoid checking each ancestor by using carats at the beginning of the node pattern. Looking at the examples in the spec, I think we could get away with only checking one or two ancestors up.

Reply:
Right, I'll update it.